common/djangoapps/student/helpers.py

```diff
@@ -230,3 +235,38 @@ def get_next_url_for_login_page(request):
         # be saved in the session as part of the pipeline state. That URL will take priority
         # over this one.
     return redirect_to
+
+
+def get_profile_visibility(user_profile, user, configuration=None):
```

Comment:
@nasthagiri Also, I tried moving it to where you suggested, but because the code there is now used in class construction for the Badging API view call permissions, it seems to cause a chicken-and-egg problem placing it where you suggest.